tests/test_asset.py

```diff
@@ -61,3 +73,194 @@ def test_public_group_asset_info():
     """Get info about a public production group."""
     info = asset.info("tiledb://TileDB-Inc/52165567-040c-4e75-bb89-a3d06017f650")
     assert info.name == "langchain_documentation_huggingface"
+
+
+@mock.patch("tiledb.object_type", return_value="array")
+@mock.patch("tiledb.cloud.array.share_array")
+def test_asset_share_array_dispatch(share_array, object_type):
+    """Dispatch to array.share_array when URI is an array."""
+    asset.share("a", "public", "read")
+    share_array.assert_called_once_with("a", "public", "read")
+
+
+@mock.patch("tiledb.object_type", return_value="group")
+@mock.patch("tiledb.cloud.groups.share_group")
+def test_asset_share_group_dispatch(share_group, object_type):
+    """Dispatch to groups.share when URI is a group."""
+    asset.share("g", "public", "read")
+    share_group.assert_called_once_with("g", "public", "read")
+
+
+@mock.patch("tiledb.object_type", return_value="group")
+@mock.patch("tiledb.cloud.groups.share_group")
+def test_asset_unshare_group_dispatch(share_group, object_type):
+    """Dispatch to groups.share when URI is a group."""
+    asset.unshare("g", "public")
+    share_group.assert_called_once_with("g", "public", [])
+
+
+@mock.patch("tiledb.object_type", return_value="array")
+@mock.patch(
+    "tiledb.cloud.array.list_shared_with", return_value=[ArraySharing(namespace="foo")]
+)
+def test_asset_list_shared_with_array_dispatch(array_sharing, object_type):
+    """Dispatch to array.list_shared_with when URI is an array."""
+    sharing = asset.list_shared_with("a")
+    assert isinstance(sharing[0], ArraySharing)
+    assert sharing[0].namespace == "foo"
+
+
+@mock.patch("tiledb.object_type", return_value="group")
+@mock.patch(
+    "tiledb.cloud.groups.list_shared_with", return_value=[GroupSharing(namespace="bar")]
+)
+def test_asset_list_shared_with_group_dispatch(group_sharing, object_type):
+    """Dispatch to groups.info when URI is a group."""
+    sharing = asset.list_shared_with("g")
+    assert isinstance(sharing[0], GroupSharing)
+    assert sharing[0].namespace == "bar"
+
+
+@pytest.mark.xfail(reason="Server error 610")
```

Comment:
Oddly there is a server error in this case, but not with the same test of a public group.